### ISSUE TITLE
💅 False positive for useConst

### ISSUE THREAD

**Issue by WandererXII** [biomejs/biome#7174]
### Environment information

```bash
CLI:
  Version:                      2.1.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux
```

### Rule name

lint/style/useConst

### Playground link

https://biomejs.dev/playground/?code=bABlAHQAIABpAG4AdABlAHIAdgBhAGwAOgAgAG4AdQBtAGIAZQByACAAfAAgAHUAbgBkAGUAZgBpAG4AZQBkADsACgAKAGMAbwBuAHMAdAAgAHIAZQBkAHIAYQB3ACAAPQAgACgAKQAgAD0APgAgAHsACgAgACAAYwBsAGUAYQByAEkAbgB0AGUAcgB2AGEAbAAoAGkAbgB0AGUAcgB2AGEAbAApADsACgAgACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAIgByAGUAZAByAGEAdwAiACkAOwAKAH0AOwAKAHIAZQBkAHIAYQB3ACgAKQA7AAoACgBpAG4AdABlAHIAdgBhAGwAIAA9ACAAcwBlAHQASQBuAHQAZQByAHYAYQBsACgAcgBlAGQAcgBhAHcALAAgADEAMAAwADAAKQA7AA%3D%3D&language=ts

### Expected result

```typescript
let interval: number | undefined;

const redraw = () => {
  clearInterval(interval);
  console.log("redraw");
};
redraw();

interval = setInterval(redraw, 1000);
```

Minimal example, `interval` is read before assignment inside a function, yet still marked as warning.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
This is an expected behavior of the current implementation: `useConst` ignores reads in inner functions. Detecting this case could require control flow analysis and function call tracking.

We could change the implementation and accept false negatives. This seems to be a better approach because the rule fix is marked as safe.
